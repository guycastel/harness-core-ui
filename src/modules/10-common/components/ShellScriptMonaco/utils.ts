/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

const EXPRESSION_START_REGEX = /<\+([A-Za-z0-9_.'"()]*?)$/

export function getStartColumnForMonacoRange(prevText?: string): number | undefined {
  if (typeof prevText !== 'string') return undefined
  const startColumnWithTriggerChar = prevText.match(EXPRESSION_START_REGEX)?.index
  if (startColumnWithTriggerChar === undefined) return undefined
  const startColumn = startColumnWithTriggerChar + 2 + 1 // 2 for trigger chars + 1 for monaco as it starts index from 1
  return startColumn
}
