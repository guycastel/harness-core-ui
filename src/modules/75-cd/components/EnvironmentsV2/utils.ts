/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { get } from 'lodash-es'
import { GitQueryParams, InfrastructureGitQueryParams } from '@modules/10-common/interfaces/RouteInterfaces'
import { EnvironmentResponseDTO, InfrastructureResponseDTO } from 'services/cd-ng'

export enum EnvironmentDetailsTab {
  CONFIGURATION = 'CONFIGURATION',
  INFRASTRUCTURE = 'INFRASTRUCTURE',
  SERVICE_OVERRIDES = 'SERVICE_OVERRIDES',
  GITOPS = 'GITOPS',
  SUMMARY = 'SUMMARY',
  REFERENCED_BY = 'REFERENCED_BY'
}

export const getRemoteEnvironmentQueryParams = (environment: EnvironmentResponseDTO): GitQueryParams => {
  if (environment?.storeType === 'REMOTE') {
    return {
      storeType: get(environment, 'storeType'),
      connectorRef: get(environment, 'connectorRef', ''),
      repoName: get(environment, 'entityGitDetails.repoName', '')
    }
  }
  return {}
}

export const getRemoteInfrastructureQueryParams = (
  infrastructure: InfrastructureResponseDTO
): InfrastructureGitQueryParams => {
  if (infrastructure?.storeType === 'REMOTE') {
    return {
      infraStoreType: get(infrastructure, 'storeType'),
      infraConnectorRef: get(infrastructure, 'connectorRef', ''),
      infraRepoName: get(infrastructure, 'entityGitDetails.repoName', '')
    }
  }
  return {}
}
