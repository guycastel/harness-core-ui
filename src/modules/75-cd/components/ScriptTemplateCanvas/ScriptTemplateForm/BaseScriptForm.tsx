/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { Formik, AllowedTypes } from '@harness/uicore'
import * as Yup from 'yup'
import type { FormikProps } from 'formik'

import { setFormikRef, StepFormikFowardRef } from '@pipeline/components/AbstractSteps/Step'
import { useStrings } from 'framework/strings'
import { LocationType } from '@cd/components/PipelineSteps/CommandScripts/CommandScriptsTypes'
import { ShellScriptFormData, variableSchema } from '@cd/components/PipelineSteps/ShellScriptStep/shellScriptTypes'
import BaseScript from '@cd/components/BaseScript/BaseScript'
import { getInitialValues } from '../../PipelineSteps/ShellScriptStep/helper'

interface ShellScriptWidgetProps {
  initialValues: ShellScriptFormData
  updateTemplate?: (data: ShellScriptFormData) => void
  onChange?: (data: ShellScriptFormData) => void
  allowableTypes: AllowedTypes
  readonly?: boolean
}

export function BaseScriptForm(
  { initialValues, updateTemplate, onChange, allowableTypes }: ShellScriptWidgetProps,
  formikRef: StepFormikFowardRef
): JSX.Element {
  const { getString } = useStrings()

  const validationSchema = Yup.object().shape({
    spec: Yup.object().shape({
      shell: Yup.string().trim().required(getString('validation.scriptTypeRequired')),
      source: Yup.object().shape({
        spec: Yup.object()
          .when(['type'], {
            is: type => {
              return type === LocationType.INLINE
            },
            then: Yup.object().shape({
              script: Yup.string().trim().required(getString('common.scriptRequired'))
            })
          })
          .when(['type'], {
            is: type => {
              return type === LocationType.HARNESS
            },
            then: Yup.object().shape({
              file: Yup.string()
                .trim()
                .required(getString('fieldRequired', { field: getString('common.git.filePath') }))
            })
          })
      }),
      environmentVariables: variableSchema(getString),
      outputVariables: variableSchema(getString)
    })
  })

  return (
    <Formik<ShellScriptFormData>
      onSubmit={/* istanbul ignore next */ submit => updateTemplate?.(submit)}
      validate={formValues => {
        onChange?.(formValues)
      }}
      formName="shellScriptForm"
      initialValues={getInitialValues(initialValues)}
      validationSchema={validationSchema}
    >
      {(formik: FormikProps<ShellScriptFormData>) => {
        setFormikRef(formikRef, formik)

        return <BaseScript formik={formik} readonly={false} allowableTypes={allowableTypes} />
      }}
    </Formik>
  )
}

export const BaseScriptWithRef = React.forwardRef(BaseScriptForm)
