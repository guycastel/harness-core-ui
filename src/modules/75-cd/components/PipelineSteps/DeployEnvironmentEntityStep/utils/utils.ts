/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { getMultiTypeFromValue, MultiTypeInputType, RUNTIME_INPUT_VALUE } from '@harness/uicore'
import { isEmpty, isEqual, isNil } from 'lodash-es'
import * as Yup from 'yup'
import type { UseStringsReturn } from 'framework/strings'
import { isValueFixed, isValueRuntimeInput } from '@common/utils/utils'
import { StageElementWrapperConfig } from 'services/pipeline-ng'
import { DeploymentStageElementConfig } from '@pipeline/utils/pipelineTypes'

import type {
  DeployEnvironmentEntityConfig,
  DeployEnvironmentEntityCustomStepProps,
  DeployEnvironmentEntityFormState
} from '../types'
import {
  processSingleEnvironmentGitOpsInitialValues,
  processSingleEnvironmentInitialValues
} from './singleEnvironment/processSingleEnvironmentInitialValues'
import {
  processSingleEnvironmentFormValues,
  processSingleEnvironmentGitOpsFormValues
} from './singleEnvironment/processSingleEnvironmentFormValues'
import { processMultiEnvironmentInitialValues } from './multiEnvironment/processMultiEnvironmentInitialValues'
import { processMultiEnvironmentFormValues } from './multiEnvironment/processMultiEnvironmentFormValues'
import { processEnvironmentGroupInitialValues } from './environmentGroup/processEnvironmentGroupInitialValues'
import { processEnvironmentGroupFormValues } from './environmentGroup/processEnvironmentGroupFormValues'

export function processInitialValues(
  initialValues: DeployEnvironmentEntityConfig,
  customStepProps: DeployEnvironmentEntityCustomStepProps,
  onUpdate?: (data: DeployEnvironmentEntityConfig) => void
): DeployEnvironmentEntityFormState {
  const gitOpsEnabled = !!customStepProps.gitOpsEnabled

  // istanbul ignore else
  if (initialValues.environment && initialValues.environment.useFromStage && !gitOpsEnabled) {
    return processSingleEnvironmentInitialValues(initialValues.environment, customStepProps)
  } else if (initialValues.environment && initialValues.environment.environmentRef) {
    if (gitOpsEnabled) {
      return processSingleEnvironmentGitOpsInitialValues(initialValues.environment)
    } else {
      // this is backwards compatibility for single env multiple clusters in the older redesign setup
      if (
        Array.isArray(initialValues.environment.gitOpsClusters) &&
        initialValues.environment.gitOpsClusters.length > 1
      ) {
        return processMultiEnvironmentInitialValues(
          {
            environments: {
              metadata: {
                parallel: false
              },
              values: [{ ...initialValues.environment }]
            }
          },
          customStepProps
        )
      }
      return processSingleEnvironmentInitialValues(initialValues.environment, customStepProps)
    }
  } else if (initialValues.environments) {
    return processMultiEnvironmentInitialValues(initialValues, customStepProps)
  } else if (initialValues.environmentGroup) {
    // This handles the migration from runtime environment group of old pipeline to post multi infra handling
    if (
      isEqual(initialValues.environmentGroup, {
        envGroupRef: RUNTIME_INPUT_VALUE,
        deployToAll: true
      })
    ) {
      const updatedInitialValues = {
        environmentGroup: {
          envGroupRef: RUNTIME_INPUT_VALUE,
          deployToAll: RUNTIME_INPUT_VALUE as any,
          environments: RUNTIME_INPUT_VALUE as any
        }
      }

      onUpdate?.(updatedInitialValues)
      return processEnvironmentGroupInitialValues(updatedInitialValues, customStepProps)
    } else if (
      // This handles the migration from 'all' environments under environment group of old pipeline to post multi infra handling
      initialValues.environmentGroup.deployToAll === true &&
      !isValueRuntimeInput(initialValues.environmentGroup.envGroupRef) &&
      isNil(initialValues.environmentGroup.envGroupRef)
    ) {
      const updatedInitialValues = {
        environmentGroup: {
          envGroupRef: initialValues.environmentGroup.envGroupRef,
          deployToAll: true,
          environments: RUNTIME_INPUT_VALUE as any
        }
      }

      onUpdate?.(updatedInitialValues)
      return processEnvironmentGroupInitialValues(updatedInitialValues, customStepProps)
    }

    return processEnvironmentGroupInitialValues(initialValues, customStepProps)
  }

  return {
    category: 'single'
  }
}

export function processFormValues(
  data: DeployEnvironmentEntityFormState,
  customStepProps: DeployEnvironmentEntityCustomStepProps
): DeployEnvironmentEntityConfig {
  const gitOpsEnabled = !!customStepProps.gitOpsEnabled
  // istanbul ignore else
  if (data.category === 'single') {
    if (gitOpsEnabled) {
      return processSingleEnvironmentGitOpsFormValues(data)
    } else {
      return processSingleEnvironmentFormValues(data, customStepProps)
    }
  } else if (data.category === 'group') {
    return processEnvironmentGroupFormValues(data, customStepProps)
  } else if (data.category === 'multi') {
    return processMultiEnvironmentFormValues(data, customStepProps)
  }
  return {}
}

export function getValidationSchema(getString: UseStringsReturn['getString'], gitOpsEnabled: boolean): Yup.MixedSchema {
  return Yup.mixed()
    .required()
    .test({
      test(valueObj: DeployEnvironmentEntityFormState): boolean | Yup.ValidationError {
        if (valueObj.category === 'single') {
          if (!valueObj.environment && isNil(valueObj.propagateFrom)) {
            return this.createError({
              path: 'environment',
              message: getString('cd.pipelineSteps.environmentTab.environmentIsRequired')
            })
          }

          if (
            !gitOpsEnabled &&
            !valueObj.infrastructure &&
            !isValueRuntimeInput(valueObj.environment) &&
            isNil(valueObj.propagateFrom)
          ) {
            return this.createError({
              path: 'infrastructure',
              message: getString('cd.pipelineSteps.environmentTab.infrastructureIsRequired')
            })
          }

          // To be put back once BE supports multi environments
          // if (gitOpsEnabled && !valueObj.cluster && !isValueRuntimeInput(valueObj.environment)) {
          //   return this.createError({
          //     path: 'cluster',
          //     message: getString('cd.pipelineSteps.environmentTab.clusterIsRequired')
          //   })
          // }
        } else if (valueObj.category === 'multi') {
          if (
            !(Array.isArray(valueObj.environments) && valueObj.environments.length) &&
            !isValueRuntimeInput(valueObj.environments as unknown as string) &&
            isEmpty(valueObj.environmentFilters?.['fixedScenario'])
          ) {
            return this.createError({
              path: 'environments',
              message: getString('cd.pipelineSteps.environmentTab.environmentsAreRequired')
            })
          }
        } else if (valueObj.category === 'group') {
          if (!valueObj.environmentGroup) {
            return this.createError({
              path: 'environmentGroup',
              message: getString('cd.pipelineSteps.environmentTab.environmentGroupIsRequired')
            })
          }
        }

        return true
      }
    })
}

export function getSelectedInfrastructuresWhenPropagating(
  value?: string,
  previousStages?: StageElementWrapperConfig[]
): string[] {
  const infrastructureDefinitions = (
    previousStages?.find(previousStage => previousStage.stage?.identifier === value)
      ?.stage as DeploymentStageElementConfig
  )?.spec?.environment?.infrastructureDefinitions

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  if (isValueRuntimeInput(infrastructureDefinitions as any)) return []

  const prevInfraId = infrastructureDefinitions?.[0].identifier
  return prevInfraId && isValueFixed(prevInfraId) ? [prevInfraId] : []
}

export function getAllFixedInfrastructures(
  data: DeployEnvironmentEntityFormState,
  environmentIdentifier: string,
  previousStages?: StageElementWrapperConfig[]
): string[] {
  if (data.propagateFrom?.value) {
    return getSelectedInfrastructuresWhenPropagating(data.propagateFrom?.value as string, previousStages)
  } else if (data.infrastructure && getMultiTypeFromValue(data.infrastructure) === MultiTypeInputType.FIXED) {
    return [data.infrastructure as string]
  } else if (
    data.infrastructures?.[environmentIdentifier] &&
    Array.isArray(data.infrastructures[environmentIdentifier])
  ) {
    return data.infrastructures[environmentIdentifier].map(infrastructure => infrastructure.value as string)
  }

  return []
}

export function getSelectedEnvironmentsWhenPropagating(
  value?: string,
  previousStages?: StageElementWrapperConfig[]
): string[] {
  const prevEnvId = (
    previousStages?.find(previousStage => previousStage.stage?.identifier === value)
      ?.stage as DeploymentStageElementConfig
  )?.spec?.environment?.environmentRef
  return prevEnvId && isValueFixed(prevEnvId) ? [prevEnvId] : []
}

export function getAllFixedEnvironments(
  data: DeployEnvironmentEntityFormState,
  previousStages?: StageElementWrapperConfig[]
): string[] {
  if (data.propagateFrom?.value) {
    return getSelectedEnvironmentsWhenPropagating(data.propagateFrom?.value as string, previousStages)
  } else if (data.environment && getMultiTypeFromValue(data.environment) === MultiTypeInputType.FIXED) {
    return [data.environment as string]
  } else if (data.environments && Array.isArray(data.environments)) {
    return data.environments.map(environment => environment.value as string)
  }

  return []
}

export function getAllFixedEnvironmentsGitDetails(
  data: DeployEnvironmentEntityFormState
): Record<string, string | undefined> {
  if (data.environment && getMultiTypeFromValue(data.environment) === MultiTypeInputType.FIXED) {
    return { [data.environment]: data.gitMetadata?.[data.environment] }
  } else if (data.environments && Array.isArray(data.environments)) {
    let gitMetadata = {}
    data.environments.forEach(environment => {
      gitMetadata = { ...gitMetadata, [environment.value as string]: data.gitMetadata?.[environment.value as string] }
    })
    return gitMetadata
  }

  return {}
}
