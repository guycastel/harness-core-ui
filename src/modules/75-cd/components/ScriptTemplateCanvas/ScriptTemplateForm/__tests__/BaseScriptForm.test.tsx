/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { MultiTypeInputType, Formik } from '@harness/uicore'
import { render, waitFor, fireEvent, queryByAttribute } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import { BaseScriptWithRef } from '../BaseScriptForm'

const defaultInitialValues = {
  identifier: 'id',
  name: 'name',
  type: '',
  spec: {}
}
const defaultInitialValuesCorrect = {
  identifier: 'id',
  name: 'name',
  type: 'PowerShell',
  spec: {
    shell: 'Bash',
    source: {
      type: 'Inline',
      spec: {
        script: 'check'
      }
    }
  }
}

const initialValues = {
  ...defaultInitialValues,
  spec: {
    shell: 'Bash',
    source: {
      spec: {
        type: 'Inline',
        script: 'echo test'
      }
    }
  }
}
const initialValuesWithExpressions = {
  ...defaultInitialValues,
  spec: {
    shell: 'Bash',
    source: {
      spec: {
        type: 'Inline',
        script: '<+spec.environmentVariable.var1>'
      }
    }
  }
}

const initialValuesWithRI = {
  ...defaultInitialValues,
  spec: {
    shell: 'Bash',
    source: {
      spec: {
        type: 'Inline',
        script: '<+input>'
      }
    }
  }
}

describe('Test BaseScriptWithRef', () => {
  test('initial render', async () => {
    const { container } = render(
      <TestWrapper>
        <BaseScriptWithRef initialValues={defaultInitialValues} allowableTypes={[]} />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })
  test('should match snapshot for BaseScriptWithRef with initial values', async () => {
    const { container, getByText } = render(
      <TestWrapper>
        <BaseScriptWithRef initialValues={initialValues} allowableTypes={[MultiTypeInputType.EXPRESSION]} />
      </TestWrapper>
    )
    await waitFor(() => getByText('cd.steps.commands.selectScriptLocation'))
    expect(container).toMatchSnapshot()
  })

  test('should match snapshot for BaseScriptWithRef with initial script values with expressions', async () => {
    const { container, getByText } = render(
      <TestWrapper>
        <BaseScriptWithRef
          initialValues={initialValuesWithExpressions}
          allowableTypes={[MultiTypeInputType.EXPRESSION]}
        />
      </TestWrapper>
    )
    await waitFor(() => getByText('cd.steps.commands.locationFileStore'))
    expect(container).toMatchSnapshot()
  })

  test('should match snapshot for BaseScriptWithRef with initial script values with runtimeInput', async () => {
    const { container } = render(
      <TestWrapper>
        <BaseScriptWithRef initialValues={initialValuesWithRI} allowableTypes={[MultiTypeInputType.RUNTIME]} />
      </TestWrapper>
    )
    const configureBtn = container.querySelector('button[id="configureOptions_spec.source.spec.script"]')

    await waitFor(() => expect(configureBtn).toBeDefined())

    expect(container).toMatchSnapshot()
  })

  test('should match snapshot for BaseScriptWithRef without initial values ', async () => {
    const onChangeMock = jest.fn()

    const { container } = render(
      <TestWrapper>
        <Formik<any> formName="test-form" initialValues={defaultInitialValuesCorrect} onSubmit={jest.fn()}>
          {formik => {
            return (
              <BaseScriptWithRef
                initialValues={defaultInitialValuesCorrect}
                allowableTypes={[]}
                ref={formik as any}
                onChange={onChangeMock}
              />
            )
          }}
        </Formik>
      </TestWrapper>
    )
    const queryByNameAttribute = (name: string): HTMLElement | null => queryByAttribute('name', container, name)
    await waitFor(() => expect(queryByNameAttribute('spec.shell')).toBeDefined())

    const radioButtons = container.querySelectorAll('input[type="radio"]')

    await fireEvent.click(radioButtons[1])
    fireEvent.input(queryByNameAttribute('spec.source.spec.script')!, {
      target: { value: 'echo Hello World' },
      bubbles: true
    })

    expect(onChangeMock).toBeCalledWith({
      identifier: 'id',
      name: 'name',
      type: 'PowerShell',
      spec: {
        executionTarget: {},
        onDelegate: true,
        shell: 'Bash',
        source: {
          type: 'Inline',
          spec: {
            script: 'echo Hello World'
          }
        }
      }
    })
  })
})
