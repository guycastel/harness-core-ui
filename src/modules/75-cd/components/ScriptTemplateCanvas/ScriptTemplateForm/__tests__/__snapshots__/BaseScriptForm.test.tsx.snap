// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BaseScriptWithRef initial render 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <div
      class="formGroup sm"
    >
      <div
        class="bp3-form-group"
      >
        <label
          class="bp3-label"
          for="spec.shell"
        >
          <span
            class="Tooltip--acenter"
            data-tooltip-id="shellScriptForm_spec.shell"
          >
            common.scriptType
          </span>
           
          <span
            class="bp3-text-muted"
          />
        </label>
        <div
          class="bp3-form-content"
        >
          <div
            class="bp3-popover-wrapper Select--main bp3-fill"
          >
            <div
              class="bp3-popover-target"
            >
              <div
                class="bp3-input-group"
              >
                <input
                  autocomplete="off"
                  class="bp3-input"
                  name="spec.shell"
                  placeholder="- common.scriptType -"
                  style="padding-right: 0px;"
                  type="text"
                  value="Bash"
                />
                <span
                  class="bp3-input-action"
                >
                  <span
                    class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                    icon="chevron-down"
                  >
                    <svg
                      data-icon="chevron-down"
                      height="14"
                      viewBox="0 0 16 16"
                      width="14"
                    >
                      <desc>
                        chevron-down
                      </desc>
                      <path
                        d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="bp3-form-group"
    >
      <label
        class="bp3-label"
        for="spec.source.type"
      >
        <span
          class="Tooltip--acenter"
          data-tooltip-id="shellScriptForm_spec.source.type"
        >
          cd.steps.commands.selectScriptLocation
        </span>
         
        <span
          class="bp3-text-muted"
        />
      </label>
      <div
        class="bp3-form-content"
      >
        <div
          class="StyledProps--main"
        >
          
          <div
            class="RadioButtonGroup--inline"
          >
            <label
              class="RadioButton--radio StyledProps--main"
            >
              <input
                class="RadioButton--input"
                name="spec.source.type"
                type="radio"
                value="Harness"
              />
              <span
                class="RadioButton--radioIcon"
              />
              cd.steps.commands.locationFileStore
            </label>
            <label
              class="RadioButton--radio StyledProps--main"
            >
              <input
                class="RadioButton--input"
                name="spec.source.type"
                type="radio"
                value="Inline"
              />
              <span
                class="RadioButton--radioIcon"
              />
              inline
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test BaseScriptWithRef should match snapshot for BaseScriptWithRef with initial script values with expressions 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <div
      class="formGroup sm"
    >
      <div
        class="bp3-form-group"
      >
        <label
          class="bp3-label"
          for="spec.shell"
        >
          <span
            class="Tooltip--acenter"
            data-tooltip-id="shellScriptForm_spec.shell"
          >
            common.scriptType
          </span>
           
          <span
            class="bp3-text-muted"
          />
        </label>
        <div
          class="bp3-form-content"
        >
          <div
            class="bp3-popover-wrapper Select--main bp3-fill"
          >
            <div
              class="bp3-popover-target"
            >
              <div
                class="bp3-input-group"
              >
                <input
                  autocomplete="off"
                  class="bp3-input"
                  name="spec.shell"
                  placeholder="- common.scriptType -"
                  style="padding-right: 0px;"
                  type="text"
                  value="Bash"
                />
                <span
                  class="bp3-input-action"
                >
                  <span
                    class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                    icon="chevron-down"
                  >
                    <svg
                      data-icon="chevron-down"
                      height="14"
                      viewBox="0 0 16 16"
                      width="14"
                    >
                      <desc>
                        chevron-down
                      </desc>
                      <path
                        d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="bp3-form-group"
    >
      <label
        class="bp3-label"
        for="spec.source.type"
      >
        <span
          class="Tooltip--acenter"
          data-tooltip-id="shellScriptForm_spec.source.type"
        >
          cd.steps.commands.selectScriptLocation
        </span>
         
        <span
          class="bp3-text-muted"
        />
      </label>
      <div
        class="bp3-form-content"
      >
        <div
          class="StyledProps--main"
        >
          
          <div
            class="RadioButtonGroup--inline"
          >
            <label
              class="RadioButton--radio StyledProps--main"
            >
              <input
                class="RadioButton--input"
                name="spec.source.type"
                type="radio"
                value="Harness"
              />
              <span
                class="RadioButton--radioIcon"
              />
              cd.steps.commands.locationFileStore
            </label>
            <label
              class="RadioButton--radio StyledProps--main"
            >
              <input
                class="RadioButton--input"
                name="spec.source.type"
                type="radio"
                value="Inline"
              />
              <span
                class="RadioButton--radioIcon"
              />
              inline
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test BaseScriptWithRef should match snapshot for BaseScriptWithRef with initial script values with runtimeInput 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <div
      class="formGroup sm"
    >
      <div
        class="bp3-form-group"
      >
        <label
          class="bp3-label"
          for="spec.shell"
        >
          <span
            class="Tooltip--acenter"
            data-tooltip-id="shellScriptForm_spec.shell"
          >
            common.scriptType
          </span>
           
          <span
            class="bp3-text-muted"
          />
        </label>
        <div
          class="bp3-form-content"
        >
          <div
            class="bp3-popover-wrapper Select--main bp3-fill"
          >
            <div
              class="bp3-popover-target"
            >
              <div
                class="bp3-input-group"
              >
                <input
                  autocomplete="off"
                  class="bp3-input"
                  name="spec.shell"
                  placeholder="- common.scriptType -"
                  style="padding-right: 0px;"
                  type="text"
                  value="Bash"
                />
                <span
                  class="bp3-input-action"
                >
                  <span
                    class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                    icon="chevron-down"
                  >
                    <svg
                      data-icon="chevron-down"
                      height="14"
                      viewBox="0 0 16 16"
                      width="14"
                    >
                      <desc>
                        chevron-down
                      </desc>
                      <path
                        d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="bp3-form-group"
    >
      <label
        class="bp3-label"
        for="spec.source.type"
      >
        <span
          class="Tooltip--acenter"
          data-tooltip-id="shellScriptForm_spec.source.type"
        >
          cd.steps.commands.selectScriptLocation
        </span>
         
        <span
          class="bp3-text-muted"
        />
      </label>
      <div
        class="bp3-form-content"
      >
        <div
          class="StyledProps--main"
        >
          
          <div
            class="RadioButtonGroup--inline"
          >
            <label
              class="RadioButton--radio StyledProps--main"
            >
              <input
                class="RadioButton--input"
                name="spec.source.type"
                type="radio"
                value="Harness"
              />
              <span
                class="RadioButton--radioIcon"
              />
              cd.steps.commands.locationFileStore
            </label>
            <label
              class="RadioButton--radio StyledProps--main"
            >
              <input
                class="RadioButton--input"
                name="spec.source.type"
                type="radio"
                value="Inline"
              />
              <span
                class="RadioButton--radioIcon"
              />
              inline
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test BaseScriptWithRef should match snapshot for BaseScriptWithRef with initial values 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <div
      class="formGroup sm"
    >
      <div
        class="bp3-form-group"
      >
        <label
          class="bp3-label"
          for="spec.shell"
        >
          <span
            class="Tooltip--acenter"
            data-tooltip-id="shellScriptForm_spec.shell"
          >
            common.scriptType
          </span>
           
          <span
            class="bp3-text-muted"
          />
        </label>
        <div
          class="bp3-form-content"
        >
          <div
            class="bp3-popover-wrapper Select--main bp3-fill"
          >
            <div
              class="bp3-popover-target"
            >
              <div
                class="bp3-input-group"
              >
                <input
                  autocomplete="off"
                  class="bp3-input"
                  name="spec.shell"
                  placeholder="- common.scriptType -"
                  style="padding-right: 0px;"
                  type="text"
                  value="Bash"
                />
                <span
                  class="bp3-input-action"
                >
                  <span
                    class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                    icon="chevron-down"
                  >
                    <svg
                      data-icon="chevron-down"
                      height="14"
                      viewBox="0 0 16 16"
                      width="14"
                    >
                      <desc>
                        chevron-down
                      </desc>
                      <path
                        d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="bp3-form-group"
    >
      <label
        class="bp3-label"
        for="spec.source.type"
      >
        <span
          class="Tooltip--acenter"
          data-tooltip-id="shellScriptForm_spec.source.type"
        >
          cd.steps.commands.selectScriptLocation
        </span>
         
        <span
          class="bp3-text-muted"
        />
      </label>
      <div
        class="bp3-form-content"
      >
        <div
          class="StyledProps--main"
        >
          
          <div
            class="RadioButtonGroup--inline"
          >
            <label
              class="RadioButton--radio StyledProps--main"
            >
              <input
                class="RadioButton--input"
                name="spec.source.type"
                type="radio"
                value="Harness"
              />
              <span
                class="RadioButton--radioIcon"
              />
              cd.steps.commands.locationFileStore
            </label>
            <label
              class="RadioButton--radio StyledProps--main"
            >
              <input
                class="RadioButton--input"
                name="spec.source.type"
                type="radio"
                value="Inline"
              />
              <span
                class="RadioButton--radioIcon"
              />
              inline
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
